.github/workflows/create-condalock-file.yml

```diff
@@ -71,8 +71,10 @@ jobs:
         run: |
           esmvaltool --help
           esmvaltool version
-      - name: Run flake8
-        run: flake8
+      - name: Run linters
+        run: |
```

Comment:
no need to run either flake8 nor pre-commit on this one since the lockfile doesn't introduce any code changes (my historical bad, sorry); if you want to be very careful maybe just run a call to eg `pre-commit --version`

Reply:
yes, good idea! No need to slow down the tests

Reply:
is that a Python `venv` or a conda one? We may want to tell it to use mamba if it's a conda one

Reply:
It says is works without any system dependencies for Python, but if I run the pre-commit hooks in a Python 3.10 conda environment that does seem to be picked up [as shown by the failing tests with the GitHub action for Python 3.10](https://github.com/ESMValGroup/ESMValCore/actions/runs/10921655486/job/30314252157), so I'm not quite sure if I understand how that works.

Reply:
OK pip is fine, like other such env-in-env applications. Was scared it's a conda fanboi :grin:

Reply:
Done in 93cfe33

Reply:
yay :beer:

Reply:
ahha - starting to make sense of this pre-commit thingie: you know it installs all stuff in ` $HOME/.cache/pre-commit` dir right?

Reply:
OK so what it does is it installs all those deps listed in the yaml file separately, each in its own venv, so they don't kill each other in a single common venv, my `.cache/pre-commit` looks like this:
```
 (esmvaltool) valeriu@valeriu-PORTEGE-Z30-C:~/ESMValTool$ ls -la ~/.cache/pre-commit/
total 72
drwxrwxr-x 11 valeriu valeriu  4096 Sep 18 16:19 .
drwx------ 16 valeriu valeriu  4096 Sep 18 16:16 ..
-rw-------  1 valeriu valeriu 20480 Sep 18 16:16 db.db
-rw-rw-r--  1 valeriu valeriu     0 Sep 18 16:13 .lock
-rw-rw-r--  1 valeriu valeriu  2284 Sep 18 16:17 pre-commit.log
-rw-rw-r--  1 valeriu valeriu   109 Sep 18 16:13 README
drwx------  4 valeriu valeriu  4096 Sep 18 16:28 repo4ep1qzk5
drwx------ 10 valeriu valeriu  4096 Sep 18 16:16 repo4uqei5t0
drwx------ 13 valeriu valeriu  4096 Sep 18 16:16 repo7jrwosxf
drwx------  7 valeriu valeriu  4096 Sep 18 16:16 repo7xmez37c
drwx------ 10 valeriu valeriu  4096 Sep 18 16:16 repoa4zz9bxd
drwx------  8 valeriu valeriu  4096 Sep 18 16:16 repoadrq8rrg
drwx------ 12 valeriu valeriu  4096 Sep 18 16:17 repoo3owvyb7
drwx------  9 valeriu valeriu  4096 Sep 18 16:16 repote4lws0p
drwx------  4 valeriu valeriu  4096 Sep 18 16:16 repotlhn3o_9
```
each of those repoxxx are seprate venvs - by the looks of it those deps need be in the master Python environment (in our case Miniforge etc) so pre-commit can grab the dependencies and put them in those little single-occupancy venvs, that's why it needed `codespell` in the Py3.10 envs in those tests

Reply:
ah there you go, explained by none other than Anthony Sottile the pre-commit developer https://stackoverflow.com/questions/66174906/pre-commit-not-finding-python-packages